doc/builders/images/binarycache.section.md

```diff
@@ -0,0 +1,32 @@
+# pkgs.mkBinaryCache {#sec-pkgs-binary-cache}
+
+`pkgs.mkBinaryCache` is a function for creating Nix flat-file binary caches. Such a cache exists as a directory on disk, and can be used as a Nix substituter by passing `--substituter file:///path/to/cache` to Nix commands.
+
+Nix packages are most commonly shared between machines using [HTTP, SSH, or S3](https://nixos.org/manual/nix/stable/package-management/sharing-packages.html), but a flat-file binary cache can still be useful in some situations. For example, you can copy it directly to another machine, or make it available on a network file system. It can also be a convenient way to make some Nix packages available inside a container via bind-mounting, when you want to efficiently run Nix builds inside a container without doing a writable mount of the host's `/nix` directory.
+
+## Example
+
+The following derivation will construct a flat-file binary cache containing the closure of `hello`.
+
+```nix
+mkBinaryCache {
+  rootPaths = [hello];
+}
+```
+
+- `rootPaths` specifies a list of root derivations. The transitive closure of these derivations will be copied into the cache.
+
+Here's an example of building and using the cache.
+
+```bash
+# Build the cache on machine "host1"
+host1> nix-build -E 'with import <nixpkgs> {}; mkBinaryCache { rootPaths = [hello]; }'
+/nix/store/cc0562q828rnjqjyfj23d5q162gb424g-binary-cache
+
+# Copy the resulting directory to "host2"
+host2> nix-build -A hello '<nixpkgs>' \
+  --option require-sigs false \
+  --option trusted-substituters file:///path/to/binary-cache \
+  --option substituters file:///path/to/binary-cache
+/nix/store/gl5a41azbpsadfkfmbilh9yk40dh5dl0-hello-2.12.1
+```
```

Comment:
I found the following form helpful for both reading the source and rendering the output appropriately. With the right CSS one can better distinguish inputs and outputs. In any case, prose should be prose and not code comments, unless there is no good way around it.

Also, please make sure the examples can actually be used in full. Don't use placeholders if it can be avoided.

````suggestion
Build the cache on one machine, `host1`:

```shellSession
nix-build -E 'with import <nixpkgs> {}; mkBinaryCache { rootPaths = [hello]; }'
```

    /nix/store/cc0562q828rnjqjyfj23d5q162gb424g-binary-cache

Copy the resulting directory to the other machine, `host2`:

```shellSession
scp result host2:/tmp/hello-cache
```

Build the derivation using the flat-file binary cache on the other machine, `host2`:
```shellSession
nix-build -A hello '<nixpkgs>' \
  --option require-sigs false \
  --option trusted-substituters file:///tmp/hello-cache \
  --option substituters file:///tmp/hello-cache
/nix/store/gl5a41azbpsadfkfmbilh9yk40dh5dl0-hello-2.12.1
```
```

Reply:
Okay, I applied this -- not sure why you're mixing in four-space code blocks since AFAIK those are identical to fenced code blocks. I'm going to change those back to normal fenced blocks.